### PR TITLE
Fix: Crash on refresh

### PR DESCRIPTION
**Pull request by joseph-belmonte**

### CODE REVIEW

File: src/utils/useStickyState.ts [hunk 1]
```diff
@@ -3,10 +3,23 @@ import React from 'react'
...
+        return new Date(value)
+      }
+      return value
+    })
   })
   React.useEffect(() => {
-    window.localStorage.setItem(key, JSON.stringify(entries))
+    window.localStorage.setItem(key, JSON.stringify(entries, (key: any, value: any) => {
+      if (value instanceof Date) {
+        return value.toISOString()
```
Comment (evening) [inline src/utils/useStickyState.ts:19]: makes sense! however i think JSON.stringify() will automatically do that

**Comment by evening** [comment]
ok i see the problem. it might be easier to get rid of the Generic part and just assume useStickyState is always Entries. i'll leave some comments but take a look at this PR https://github.com/evening/expiration-tracker/pull/14

File: src/utils/useStickyState.ts [hunk 2]
```diff
@@ -3,10 +3,23 @@ import React from 'react'
 function useStickyState<T> (key: string) {
   const [entries, setEntries] = React.useState<T>(() => {
     const stickyValue = window.localStorage.getItem(key)
-    return stickyValue !== null ? JSON.parse(stickyValue) : []
+    if (stickyValue === null) {
+      return []
+    }
+    return JSON.parse(stickyValue, (key: any, value: any) => {
+      if (value !== null && value !== undefined && typeof value === 'string' && value.match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/)) {
```
Comment (evening) [inline src/utils/useStickyState.ts:10]: instead of validating it's a date with regex, i think you could use Date.parse() and check if it's NaN

**Comment by joseph-belmonte** [comment]
Fixed by changing the stickystate file, see: https://github.com/evening/expiration-tracker/pull/14

**Pull request closed without merging**